File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    
    //sort the arry and count the overlap element 
    intervals.sort((a,b)=> a[1] - b[1]);
    // console.log(intervals);
    let pre = 0,count = 0;
    for(let i = 1; i < intervals.length; i++){
        
        if(intervals[i][0] < intervals[pre][1])
            count++;
        else
            pre = i;
    }
    
    return count;
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    static bool cmp(vector<int>&a,vector<int>&b){
        return a[1]<b[1];
    }
    
    int eraseOverlapIntervals(vector<vector<int>>& inter) {
        sort(inter.begin(),inter.end(),cmp);
        int count=0,overlapEndpoint=0;
        for(int i = 1; i<inter.size(); i++){
            if(inter[i][0] < inter[overlapEndpoint][1])
                count++;
            else
                overlapEndpoint = i;
        }
    return count;
    }
};